python/salpytools/salpylib.py

```diff
@@ -398,17 +419,18 @@ def run_Command(self):
             time.sleep(self.tsleep)
         return
 
-    def getCurrent(self):
+    def getCurrent(self,getNone=False):
         if len(self.myDatalist) > 0:
             Current = self.myDatalist[-1]
             self.newTelem = False
             self.newEvent = False
         else:
-            # Current = None
-            # For now we're passing the empty value of the object, we might want to revise this in the future
-            LOGGER.info(
-                "WARNING: No value received for: '{}' yet, sending empty object anyway".format(self.topic))
-            Current = self.myData
+            if getNone:
+                Current = None
+            else:
+                Current = self.myData
+            msg = "No value received for topic: '{}'".format(self.topic)
+            LOGGER.warning(msg)
```

Comment:
Strings within the lsst.log.warn() methods (usage suggested above) using lazy evaluation, and will only be evaluated if the log record is to be printed, so the two lines should be merged.

Reply:
Thanks Steve, will change the evaluation method as you suggest.